packages/transformers/babel/src/config.js

```diff
@@ -1,141 +1,191 @@
 // @flow
-import type {MutableAsset} from '@parcel/types';
-import getBabelRc from './babelrc';
-import getEnvConfig from './env';
-import getJSXConfig from './jsx';
-import getFlowConfig from './flow';
+import nullthrows from 'nullthrows';
 import path from 'path';
+import {loadPartialConfig, createConfigItem} from '@babel/core';
+
+import {IConfig} from '@parcel/types';
+import {md5FromObject} from '@parcel/utils';
+
+import getEnvOptions from './env';
+import getJSXOptions from './jsx';
+import getFlowOptions from './flow';
+import getTypescriptOptions from './typescript';
+import type {BabelConfig} from './types';
 
 const TYPESCRIPT_EXTNAME_RE = /^\.tsx?/;
-const NODE_MODULES = `${path.sep}node_modules${path.sep}`;
-
-export default async function getBabelConfig(asset: MutableAsset) {
-  // Consider the module source code rather than precompiled if the resolver
-  // used the `source` field, or it is not in node_modules.
-  let pkg = await asset.getPackage();
-  let isSource =
-    !!(
-      pkg &&
-      pkg.source &&
-      (await asset.fs.realpath(asset.filePath)) !== asset.filePath
-    ) || !asset.filePath.includes(NODE_MODULES);
-
-  // Try to resolve a .babelrc file. If one is found, consider the module source code.
-  let babelrc = await getBabelRc(asset, pkg, isSource);
-  isSource = isSource || !!babelrc;
-
-  let result = {};
-  mergeConfigs(result, babelrc);
-
-  // Typescript must use the plugin directly (not the typescript preset) and must
-  // come before preset env, otherwise proposals and nonstandard syntax is not
-  // transformed in time.
-  if (path.extname(asset.filePath).match(TYPESCRIPT_EXTNAME_RE)) {
-    let hasTypescript =
-      babelrc &&
-      (hasPlugin(babelrc.config.presets, [
-        '@babel/typescript',
-        '@babel/preset-typescript'
-      ]) ||
-        hasPlugin(babelrc.config.plugins, [
-          '@babel/transform-typescript',
-          '@babel/plugin-transform-typescript'
-        ]));
-
-    if (!hasTypescript) {
-      mergeConfigs(result, {
-        internal: true,
-        babelVersion: 7,
-        config: {
-          plugins: [
-            [
-              require('@babel/plugin-transform-typescript'),
-              {isTSX: path.extname(asset.filePath) === '.tsx'}
-            ]
-          ]
-        }
+const BABEL_TRANSFORMER_DIR = path.dirname(__dirname);
+
+export async function load(config: IConfig) {
+  let partialConfig = loadPartialConfig({filename: config.searchPath});
+  if (partialConfig && partialConfig.hasFilesystemConfig()) {
+    let {babelrc, config: configjs} = partialConfig;
+    let {canBeRehydrated, dependsOnRelative, dependsOnLocal} = getStats(
+      partialConfig.options
+    );
+
+    // babel.config.js files get required by @babel/core so there's no use in including it for watch mode invalidation
+    if (babelrc != null && configjs == null) {
```

Comment:
what about `.babelrc.js`? Does that apply to the `babelrc` or `config` field?

Reply:
Mmm, that's a good question. I was thinking that babelrc was specifically the old json style and config was for the new js style. I'll double check though.

Reply:
`.babelrc.js` is definitely still supported: https://babeljs.io/docs/en/configuration#babelrcjs

Reply:
Yeah, I was just thinking that the `babelrc` property returned from `loadPartialConfig` was for json and the property`config` was for js. Looks like it uses the `babelrc` field for `.babelrc.js` though 😕 Ugh, is the distinction just that `babel.config.js` behaves differently because it's expected to be at the root? Guess I'll have to figure out a different way to determine if a js file was used.

Reply:
I am now checking the ext of babelrc to see if it is a js file.